clang/include/clang/AST/StmtOpenMP.h

```diff
@@ -6106,6 +6106,8 @@ class OMPTeamsGenericLoopDirective final : public OMPLoopDirective {
 class OMPTargetTeamsGenericLoopDirective final : public OMPLoopDirective {
   friend class ASTStmtReader;
   friend class OMPExecutableDirective;
+  /// true if loop directive's associated loop can be a parallel for.
+  bool CanBeParallelFor = false;
```

Comment:
Do you still need to this new field, taking into account that it does not affect sema anymore, only codegen? You don't need to store this flag, instead you can call teamsLoopCanBeParallelFor() directly in codegen

Reply:
Hi Alexey - thanks for the comment.  A clarification, as I'm not sure exactly what you're referring to.

So currently, when OMPTargetTeamsGenericLoopDirective is created in SemaOpenMP.cpp, teamsLoopCanBeParallelFor(AStmt) is called as an argument to the create, the result of which is stored in CanBeParallelFor. Later, when the directive is seen in CodeGen/CGStmtOpenMP.cpp, the boolean CanBeParallelFor (via the function canBeParallelFor()) is checked to determine how to emit the directive (parallel or distribute).

Are you saying that instead of checking whether the loop can be parallel while we're in Sema, and saving that value when we create the target teams loop directive, that we should determine this through a call to the Sema function teamsLoopCanBeParallelFor() while in CodeGen?

Reply:
I'm saying that you can move this function to the codegen and call it in the codegen

Reply:
Ok. Thanks, Alexey.

Reply:
I don't think it is possible to have the analysis in Sema and not use a flag here.

The two options we have are:
1. Do the analysis in Sema and have the flag and then read the flag in CG.
2. Have the analysis in CG and then there's no reason to pass anything around and CG can call the function when needed.

There is a 3rd hybrid way to do this where this function is moved back into CG:
```
bool Sema::teamsLoopCanBeParallelFor(Stmt *AStmt) {
  TeamsLoopChecker Checker(*this);
  Checker.Visit(AStmt);
  return Checker.teamsLoopCanBeParallelFor();
}
```

But then I don't know how you can call the TeamsLoopChecker which lives in Sema.